src/util/thread.h

```diff
@@ -21,6 +22,11 @@ void ThreadSetInternalName(std::string&&);
 //! logging.
 const std::string& ThreadGetInternalName();
 
+/**
+ * A wrapper for do-something-once thread functions.
```

Comment:
Based on the circular dependency and what this function does (Run a thread with log statements), maybe this is better suited in `logging.h`? No strong opinion, so feel free to not move this at all.

Reply:
Maybe keep `threadnames.{h,cpp}` _and_ move `TraceThread()` to new `thread.{h,cpp}`?

Reply:
Sure if that works. Cutting down the util/system bloat can't hurt.

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/19064#issuecomment-636361588).